lib/logstash-logger/encoder.rb

```diff
@@ -0,0 +1,53 @@
+module LogStashLogger
+  module Encoder
+    DEFAULT_ENCODER = :to_json
+
+    autoload :JsonGenerate, 'logstash-logger/encoder/json_generate'
+    autoload :ToJson, 'logstash-logger/encoder/to_json'
+
+    def self.instance
```

Comment:
This could potentially be simplified by using Ruby standard library's `Singleton` module.